src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -607,15 +606,28 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         String snapshotQuery = queryModel.getQuery();
         Query query = Query.parse(snapshotQuery);
         String datasetName = query.getDatasetName();
-        createSnapshotTablesFromDatasetAsset(datasetName, assetName, snapshotSource, snapshot);
+        Dataset queryDataset = datasetService.retrieveByName(datasetName);
```

Comment:
Is this not the same as `dataset` above? What is the difference between `datasetName` and `requestContents.getDatasetName`? And with my PR, should this change to using the datasetId that gets passed?

Reply:
I've added a ticket for us to revisit this in a separate PR. https://broadworkbench.atlassian.net/browse/DC-1167